Airwallex/Card/Internal/AWXCardViewModel.swift

```diff
@@ -0,0 +1,222 @@
+    public var isBillingInformationRequired: Bool {
+        session?.isBillingInformationRequired ?? false
+    }
+
+    public var isCardSavingEnabled: Bool {
+        session is AWXOneOffSession && session?.customerId() != nil
+    }
+
+    public var initialBilling: AWXPlaceDetails {
+        session?.billing ?? AWXPlaceDetails()
+    }
+
+    public var selectedCountry: AWXCountry?
+
+    var session: AWXSession?
+    var supportedCardSchemes: [AWXCardScheme] = []
+
+    public init(session: AWXSession, supportedCardSchemes: [AWXCardScheme]) {
+        super.init()
+        self.session = session
+        selectedCountry = AWXCountry.countryWithCode(session.billing?.address?.countryCode ?? "")
+        isReusingShippingAsBillingInformation =
+            session.billing != nil && session.isBillingInformationRequired
+        self.supportedCardSchemes = supportedCardSchemes
+    }
+
+    public func setReusesShippingAsBillingInformation(_ reusesShippingAsBillingInformation: Bool)
+        throws
+    {
+        if reusesShippingAsBillingInformation && session?.billing == nil {
+            throw NSError.errorForAirwallexSDK(with: -1, localizedDescription: NSLocalizedString("No shipping address configured.", comment: ""))
+        } else {
+            isReusingShippingAsBillingInformation = reusesShippingAsBillingInformation
+        }
+    }
+
+    public func makeBilling(
+        firstName: String,
+        lastName: String,
+        email: String,
+        phoneNumber: String,
+        state: String,
+        city: String,
+        street: String,
+        postcode: String
+    ) -> AWXPlaceDetails {
+        if isReusingShippingAsBillingInformation, let billing = session?.billing {
+            return billing
+        }
+
+        let place = AWXPlaceDetails()
+        place.firstName = firstName
+        place.lastName = lastName
+        place.email = email
+        place.phoneNumber = phoneNumber
+
+        let address = AWXAddress()
+        address.countryCode = selectedCountry?.countryCode ?? ""
+        address.state = state
+        address.city = city
+        address.street = street
+        address.postcode = postcode
+
+        place.address = address
+        return place
+    }
+
+    public func makeCard(
+        name: String,
+        number: String,
+        expiry: String,
+        cvc: String
+    ) -> AWXCard {
+        let dates = expiry.split(separator: "/")
+        let card = AWXCard()
+        card.name = name
+        card.number = number.replacingOccurrences(of: " ", with: "")
+        card.expiryYear = "20\(dates.last ?? "")"
+        card.expiryMonth = "\(dates.first ?? "")"
+        card.cvc = cvc
+
+        return card
+    }
+
+    public func makeDisplayedCardBrands() -> [Any] {
+        var cardBrands = [Any]()
+        for brand in AWXBrandType.allCases {
+            for cardScheme in supportedCardSchemes {
+                if cardBrandFromCardScheme(cardScheme).rawValue == brand.rawValue {
```

Comment:
Can this efficiency be improved? We don't have to compare the rest of the array once a matched brand is found.

Reply:
Can not  pass an [AWXBrandType] to OC but I think [Int] is much better than [Any].